src/app/components/Forms/EditForm.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react'
+import FormInput from '../FormInput/FormInput'
+import PopupForm from './PopupForm'
+
+type ContentType = {
+  title?: string
+  description?: string
+}
+
+type EditFormProps = {
+  onSubmit: (payload: ContentType) => void
+  onCancel: () => void
+  content: ContentType
+}
+
+const MAX_LENGTHS = {
+  need: 80,
+  title: 40,
+  description: 144,
+}
+
+type DEFAULT = {
+  description: string
+}
+
+export default function EditForm({
+  onSubmit,
+  onCancel,
+  content,
+}: EditFormProps): JSX.Element {
+  const [data, setData] = useState<ContentType | DEFAULT>(content)
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+
+    data && onSubmit(data)
+  }
+
+  function handleCancel() {
+    setData({ description: '' })
+    onCancel()
+  }
+
+  const ContentKeys = Object.keys(data)
+
+  return (
+    <PopupForm
+      SubmitText="Edit"
+      onSubmit={handleSubmit}
+      onCancel={handleCancel}
+    >
+      {ContentKeys.map(key => {
+        const value = data[key as keyof typeof data]
+        const diff =
+          value !== undefined
+            ? MAX_LENGTHS[key as keyof typeof content] - value.length
+            : Infinity
+
+        return (
+          <FormInput
+            key={key}
+            type={key === 'description' ? 'textArea' : 'text'}
```

Comment:
First, it's very nice to see that you dynamically generate your form.
To make the form even more independent, you should not ask which `key` you have.
Instead, I would add these data as property.